astropy/tests/pytest_plugins.py

```diff
@@ -190,21 +191,11 @@ def collect(self):
                     yield doctest_plugin.DoctestItem(
                         test.name, self, runner, test)
 
-        # This is for py.test prior to 2.4.0
-        def runtest(self):
-            return
-
-    class DocTestTextfilePlus(doctest_plugin.DoctestTextfile):
+    class DocTestTextfilePlus(doctest_plugin.DoctestItem, pytest.Module):
         def runtest(self):
             # satisfy `FixtureRequest` constructor...
             self.funcargs = {}
-            try:
-                self._fixtureinfo = doctest_plugin.FuncFixtureInfo((), [], {})
-                fixture_request = doctest_plugin.FixtureRequest(self)
-            except AttributeError:  # pytest >= 2.8.0
-                python_plugin = config.pluginmanager.getplugin('python')
-                self._fixtureinfo = python_plugin.FuncFixtureInfo((), [], {})
-                fixture_request = python_plugin.FixtureRequest(self)
+            fixture_request = doctest_plugin._setup_fixtures(self)
```

Comment:
Staring at the `_pytest/doctest.py` source code... Which also convinced me the whole thing can be done more elegantly (probably by subclassing `DoctestItem`, as you suggested elsewhere).

But happy enough if things work for now...